src/rewriter.fs

```diff
@@ -327,11 +333,24 @@ let private computeAllDependencies code =
     let fct = code |> List.choose (function
         | Function(fct, block) as f -> Some (fct.fName, block, f)
         | _ -> None)
-    let deps = fct |> List.map (fun (name, block, f) ->
-        let dep = computeDependencies block
-                  |> List.filter (fun name -> fct |> List.exists (fun (x,_,_) -> name = x.Name))
-        name, dep, f)
-    deps
+    let nodes = fct |> List.map (fun (name, block, f) ->
+        let callees = computeDependencies block
+                      |> List.filter (fun name -> fct |> List.exists (fun (x,_,_) -> name = x.Name))
+        { CallGraphNode.func = f; fName = name; callees = callees })
+    nodes
+
+let removeUnused code =
+    let nodes = computeAllDependencies code
+    let isUnused name =
+        name <> "main" &&
+            not options.preserveExternals &&
+            not options.preserveAllGlobals &&
+            not (options.noRenamingList |> List.contains name) &&
+            not (nodes |> List.exists (fun n -> n.callees |> List.contains name))
+    let unused = nodes |> List.filter (fun node -> isUnused node.fName.Name)
+    code |> List.filter (function
+        | Function _ as t -> not (unused |> List.map (fun node -> node.func) |> List.contains t)
```

Comment:
nit. I think you can you `List.exists`

Reply:
or simplify the code more with `List.forall`.

But it's probably better to create a set first and avoid the n^2 complexity, e.g. (untested):

```fsharp
let unused = set [for node in nodes if isUnused node.fName.Name -> node.func]
```